src/rgw/rgw_civetweb.cc

```diff
@@ -9,22 +9,31 @@
 #include "civetweb/civetweb.h"
 #include "rgw_civetweb.h"
 
-
+#define dout_context g_ceph_context
 #define dout_subsys ceph_subsys_rgw
 
 size_t RGWCivetWeb::write_data(const char *buf, const size_t len)
 {
+  size_t off = 0;
   auto to_sent = len;
   while (to_sent) {
-    const int ret = mg_write(conn, buf, len);
+    const int ret = mg_write(conn, buf + off, to_sent);
```

Comment:
looks right to me, but can we drop the dout messages?

Reply:
ok